packages/react-router/lib/router/utils.ts

```diff
@@ -192,6 +193,10 @@ export interface ShouldRevalidateFunction {
   (args: ShouldRevalidateFunctionArgs): boolean;
 }
 
+export interface LayoutComponentProps {
+  children: ReactNode
+}
```

Comment:
Using `PropsWithChildren` would be more correct I think.

```suggestion
export type LayoutComponentProps = PropsWithChildren;
```

Reply:
I was thinking of leaving it as this in case the team decides in the future to add more props, I can change it as well, no problem.

What do you think?

Reply:
If the team want to add more props, they can still do

```ts
export type LayoutComponentProps = PropsWithChildren & {
  extraProp: type;
};
```

Reply:
PropsWithChildren also takes a generic to add extra props.

Reply:
Thanks, it's rare that I have a TIL in react. Added the changes!